src/Mod/Arch/ArchFloor.py

```diff
@@ -211,24 +307,60 @@ def removeObject(self,child):
 
 
 class _ViewProviderFloor:
-
-    "A View Provider for the Floor object"
+    """A View Provider for the Floor object."""
 
     def __init__(self,vobj):
+        """Initialises the floor view provider.
```

Comment:
I'm not sure if you have tested this with Sphinx, but I remember reading that the documentation to a class should be provided only in one section, either after the class definition after `class`, or inside the `__init__` method.

I use the first option, I place the entire docstring of the `__init__` method below the `class` definition, and then I do not provide a docstring for `__init__` explicitly. This method isn't considered like a separate method, more like part of the normal initialization of the class, so this is why its documentation can be included in the `class` itself.

Reply:
Sphinx supports having the docstrings in either the __init__, the class defintion, or both. To stick with convention I've moved the __init__ docstrings to under the class definitions.

Resolved with e87f1732f538f2ca62b1ef1ce4cef8908ee5bd94